File: openadmin/spec/spec.py
# SPDX-FileCopyrightText: 2026 OpenAdmin
#
# SPDX-License-Identifier: AGPL-3.0-or-later


from typing import NotRequired, TypedDict

from .section import Section


class Spec(TypedDict):
    version: str
    name: str
    description: NotRequired[str | None]
    sections: list[Section]

File: client/src/schemas/spec.ts
// SPDX-FileCopyrightText: 2026 OpenAdmin
//
// SPDX-License-Identifier: AGPL-3.0-or-later

import { z } from "zod"
import { sectionSchema } from "./section"

export const specSchema = z.object({
	version: z.string(),
	name: z.string(),
	id: z.string(),
	description: z.string().nullable().optional(),
	sections: z.array(sectionSchema),
})

export type Spec = z.infer<typeof specSchema>
